src/lib/__tests__/InstantSearch-test.js

```diff
@@ -281,7 +281,9 @@ describe('InstantSearch lifecycle', () => {
       });
 
       it('calls helper.search()', () => {
-        expect(search.helper.search).toHaveBeenCalledTimes(1);
+        expect(
+          search.helper.searchOnlyWithDerivedHelpers
```

Comment:
The tests are brittle and rely on implementation details. I didn't change them to avoid too many changes but we'll have to reconsider this practice it makes them a bit useless for such changes. I don't have a good answer but it's something that we have to investigate (rely on something that is not as much tied to our core logic, client?).

Reply:
testing the client indeed seems more our contract now, especially considering we might want to integrate the helper more closely